### PR TITLE
Restore functional test

### PR DESCRIPTION
**Pull request by eugenepeh**
Continuing work of #91

Proposed message:

```
The assertion of javascript files has been omitted due to incorrect
keyword specified in file extension check.

This causes functional test passes irregardless of result which
doesn't provide a clear indication on the state of our project.

Let's fix the incorrect extension, and also all the OS and machine
ambiguities to restore our functional test.
```

### CODE REVIEW

File: src/functional/java/reposense/Entry.java [hunk 1]
```diff
@@ -44,17 +44,18 @@ private void verifyAllJson(File expectedDirectory, String actualRelative) {
...
+                String relativeDirectory = file.getAbsolutePath()
+                        .replace("\\", "/").split("expected/")[1];
                 assertJson(file, relativeDirectory, actualRelative);
             }
         }
     }
 
     private void assertJson(File expectedJson, String expectedPosition, String actualRelative) {
-        File actual = new File(actualRelative + expectedPosition);
+        File actual = new File(actualRelative + "/" + expectedPosition);
```
Comment (damithc) [inline src/functional/java/reposense/Entry.java]: Should we use an OS-independent method such as `File.separator` instead of `/`?

Comment (eugenepeh) [reply src/functional/java/reposense/Entry.java]: I'm thinking of switching to [Paths](https://docs.oracle.com/javase/10/docs/api/java/nio/file/Paths.html) once I cleared this up. For those that are going to be printed to json files, we still need to use a fixed standard e.g. `/` though.

Comment (eugenepeh) [reply src/functional/java/reposense/Entry.java]: On second thought, too many File occurrence to fix at the moment. Will change according to suggestion.

**Comment by eugenepeh** [comment]
@damithc Prof, for your final review.

**Comment by damithc** [comment]
Suggested commit message:
```
1. Javascript files are being omitted from functional tests 
due to incorrect keyword specified in file extension check. 
Let's correct the extension.

2. The output of the analysis vary (cosmetically) based on
the OS and the computer. This means it is impossible to
specify the 'expected' output for functional tests. 
Lets tweak the code to produce the exact output same  
output irrespective of OS and the computer.
```

**Pull request merged**